signer/src/bitcoin/validation.rs

```diff
@@ -1318,6 +1390,262 @@ mod tests {
+            // assessed fee.
+            max_fee: TX_FEE.to_sat(),
+            // This does not matter during validation.
+            recipient: ScriptBuf::new().into(),
+            // This needs to be WITHDRAWAL_BLOCKS_WAIT less than the
+            // chain_tip_height.
+            bitcoin_block_height: 0,
+        },
+        // This is part of sBTC consensus.
+        chain_tip_height: WITHDRAWAL_BLOCKS_WAIT,
+        // This is set by Emily.
+        limits: SbtcLimits::new_per_withdrawal(Amount::ONE_BTC.to_sat()),
+        status: WithdrawalValidationResult::Ok,
+    } ; "happy-path-ok")]
+    #[test_case(WithdrawalReportErrorMapping {
+        report: WithdrawalRequestReport {
+            status: WithdrawalRequestStatus::Confirmed,
+            id: QualifiedRequestId {
+                request_id: 0,
+                txid: StacksTxId::from([0; 32]),
+                block_hash: StacksBlockHash::from([0; 32]),
+            },
+            is_accepted: Some(true),
+            amount: Amount::ONE_BTC.to_sat() + 1,
+            max_fee: TX_FEE.to_sat(),
+            recipient: ScriptBuf::new().into(),
+            bitcoin_block_height: 0,
+        },
+        status: WithdrawalValidationResult::AmountTooHigh,
+        chain_tip_height: WITHDRAWAL_BLOCKS_WAIT,
+        limits: SbtcLimits::new_per_withdrawal(Amount::ONE_BTC.to_sat()),
+    } ; "amount-too-high")]
+    #[test_case(WithdrawalReportErrorMapping {
+        report: WithdrawalRequestReport {
+            status: WithdrawalRequestStatus::Confirmed,
+            id: QualifiedRequestId {
+                request_id: 0,
+                txid: StacksTxId::from([0; 32]),
+                block_hash: StacksBlockHash::from([0; 32]),
+            },
+            is_accepted: Some(true),
+            amount: WITHDRAWAL_DUST_LIMIT - 1,
+            max_fee: TX_FEE.to_sat(),
+            recipient: ScriptBuf::new().into(),
+            bitcoin_block_height: 0,
+        },
+        chain_tip_height: WITHDRAWAL_BLOCKS_WAIT,
+        limits: SbtcLimits::new_per_withdrawal(Amount::ONE_BTC.to_sat()),
+        status: WithdrawalValidationResult::AmountIsDust,
+    } ; "amount-is-dust")]
+    #[test_case(WithdrawalReportErrorMapping {
+        report: WithdrawalRequestReport {
+            status: WithdrawalRequestStatus::Confirmed,
+            id: QualifiedRequestId {
+                request_id: 0,
+                txid: StacksTxId::from([0; 32]),
+                block_hash: StacksBlockHash::from([0; 32]),
+            },
+            is_accepted: Some(true),
+            amount: Amount::ONE_BTC.to_sat(),
+            max_fee: TX_FEE.to_sat() - 1,
+            recipient: ScriptBuf::new().into(),
+            bitcoin_block_height: 0,
+        },
+        chain_tip_height: WITHDRAWAL_BLOCKS_WAIT,
+        limits: SbtcLimits::new_per_withdrawal(Amount::ONE_BTC.to_sat()),
+        status: WithdrawalValidationResult::FeeTooHigh,
+    } ; "fee-too-high")]
```

Comment:
(related to the comment above) I would add a test for `max_fee: TX_FEE.to_sat()` and `amount: WITHDRAWAL_DUST_LIMIT` (as `WITHDRAWAL_DUST_LIMIT < TX_FEE`)

Reply:
Also done in https://github.com/stacks-network/sbtc/pull/1352/commits/3c088943b7d4cd355cb2dca1c90f16d8a30f4905